Sources/StreamChat/Workers/ChatClientUpdater.swift

```diff
@@ -93,22 +100,28 @@ class ChatClientUpdater {
             switch $0 {
             case let .success(newToken):
                 do {
-                    try self.prepareEnvironment(
-                        userInfo: userInfo,
-                        newToken: newToken
-                    )
-
                     // We manually change the `connectionStatus` for passive client
                     // to `disconnected` when environment was prepared correctly
                     // (e.g. current user session is successfully restored).
                     if !self.client.config.isClientInActiveMode {
                         self.client.connectionStatus = .disconnected(error: nil)
                     }
-
-                    self.connect(
+                    
+                    try self.prepareEnvironment(
                         userInfo: userInfo,
-                        completion: completion
-                    )
+                        newToken: newToken
+                    ) { error in
```

Comment:
`self.prepareEnvironment` should run regardless of `active/passive` client's mode because we assign a token in there. If client is passive, it's still allowed to send API calls which won't pass through if there's no token